src/main/resources/com/google/api/codegen/csharp/gapic_resourcenames.snip

```diff
@@ -0,0 +1,307 @@
+@extends "csharp/common.snip"
+
+@snippet generate(resourceNames)
+    {@renderFileHeader(resourceNames.fileHeader)}
+
+    namespace {@resourceNames.fileHeader.packageName}
+    {
+        public interface IResourceName{}// TODO: This will be in GAX
+        public interface ISingleResourceName:IResourceName{} // TODO: Is this required?
+
+        public sealed class UnknownResourceName : IResourceName // TODO: This will be in GAX
```

Comment:
Hmm... this doesn't guarantee that it's a valid resource name at all, of course, in terms of reserved characters etc. We can tighten that aspect up later, of course - and make sure that the XML documentation only guarantees that TryParse will return true when it's a valid resource name, etc.